nitro-root-enclave-server/src/main.rs

```diff
@@ -0,0 +1,328 @@
+    let (token, _public_key) = match received_message {
+        NitroRootEnclaveMessage::TokenData(tok, pubkey) => (tok, pubkey),
+        _ => return Err(NitroServerError::InvalidRootEnclaveMessage),
+    };
+
+    println!(
+        "nitro-root-enclave-server::native_attestation posting native_attestation_token to tabasco"
+    );
+    post_native_attestation_token(tabasco_url, &token, device_id)?;
+    println!("nitro-root-enclave-server::native_attestation returning Ok");
+    return Ok(nre_enclave);
+}
+
+fn post_native_attestation_token(
+    tabasco_url: &str,
+    token: &Vec<u8>,
+    device_id: i32,
+) -> Result<(), NitroServerError> {
+    println!("nitro-root-enclave-server::post_native_attestation_token started");
+    let serialized_tabasco_request =
+        colima::serialize_native_psa_attestation_token(token, device_id)
+            .map_err(|err| NitroServerError::Colima(err))?;
+    let encoded_str = base64::encode(&serialized_tabasco_request);
+    let url = format!("{:}/Nitro/AttestationToken", tabasco_url);
+    println!(
+        "nitro-root-enclave-server::post_native_attestation_token posting to URL{:?}",
+        url
+    );
+    let response = post_buffer(&url, &encoded_str)?;
+
+    println!(
+        "nitro-root-enclave-server::post_psa_attestation_token received buffer:{:?}",
+        response
+    );
+    return Ok(());
+}
+
+fn fetch_firmware_version(nre_enclave: &NitroEnclave) -> Result<String, NitroServerError> {
+    println!("nitro-root-enclave-server::fetch_firmware_version started");
+
+    let firmware_version: String = {
+        let message = NitroRootEnclaveMessage::FetchFirmwareVersion;
+        let message_buffer =
+            bincode::serialize(&message).map_err(|err| NitroServerError::Bincode(*err))?;
+        println!(
+            "SinaloaNitro::Fetch_firmware_version sending message_buffer:{:?}",
+            message_buffer
+        );
+        nre_enclave.send_buffer(&message_buffer)?;
+
+        let returned_buffer = nre_enclave.receive_buffer()?;
+        let response: NitroRootEnclaveMessage = bincode::deserialize(&returned_buffer)
+            .map_err(|err| NitroServerError::Bincode(*err))?;
+        match response {
+            NitroRootEnclaveMessage::FirmwareVersion(version) => version,
+            _ => return Err(NitroServerError::InvalidRootEnclaveMessage),
+        }
+    };
+    println!("nitro-root-enclave-server::fetch_firmware_version finished");
+    return Ok(firmware_version);
+}
+
+fn send_start(
+    url_base: &str,
+    protocol: &str,
+    firmware_version: &str,
+) -> Result<(Vec<u8>, i32), NitroServerError> {
+    let tabasco_response = send_tabasco_start(url_base, protocol, firmware_version)?;
+    match tabasco_response.has_psa_attestation_init() {
+        false => {
+            return Err(NitroServerError::InvalidProtoBufMessage);
+        }
+        true => {
+            let (challenge, device_id) =
```

Comment:
I think this can be refactored to `if send_tabasco_start(url_base, protocol, firmware_version)?.has_psa_attestation_init() { ... } else { ... }`.